platform/android/api/jni_singleton.h

```diff
@@ -219,6 +219,7 @@ class JNISingleton : public Object {
 	}
 
 	void add_signal(const StringName &p_name, const Vector<Variant::Type> &p_args) {
+		ERR_FAIL_COND_MSG(p_args.size() > 5, "Error: Currently only a maximum of 5 parameters are supported. Signal name: " + p_name);
```

Comment:
Is there reason to limit it to 5 parameters, if it can be used with more than 5 it should construct `MethodInfo` dynamically to support arbitrary number of arguments, something like:

```c++
MethodInfo mi;
mi.name = p_name;
for (int i = 0; i < p_args.size(); i++) {
    mi.arguments.push_back(PropertyInfo(p_args[i], "arg" + itos(i)));
}
ADD_SIGNAL(mi);
```

Reply:
I think your solution is better.

Reply:
Is there a limit to the number of gdscript function parameters? If so, it is best to add an error log.